### PR TITLE
Add prism element type to structure mesh generator tool.

### PR DESCRIPTION
**Pull request by TomFischer**
This allows to generate tests for the prism element.

1. [x] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.3.1)
2. [x] Tests covering your feature were added?
  The feature is tested already in `Tests/MeshLib/TestMeshGenerator.cpp`
3. [x] Any new feature or behavior change was documented?
  Command line help is updated.

### CODE REVIEW

File: MeshLib/MeshEnums.cpp [hunk 1]
```diff
@@ -91,35 +93,35 @@ std::string MeshElemType2StringShort(const MeshElemType t)
...
     {
         return MeshElemType::TRIANGLE;
     }
-    if (s == "tet" || s == "Tetrahedron")
+    if (boost::iequals(s, "tet") || boost::iequals(s, "Tetrahedron"))
     {
         return MeshElemType::TETRAHEDRON;
     }
-    if (s == "pris" || s == "Prism")
+    if (boost::iequals(s, "pris") || boost::iequals(s, "prism"))
```
Comment (endJunction) [inline MeshLib/MeshEnums.cpp:120]:
```suggestion
    if (boost::iequals(s, "pris") || boost::iequals(s, "Prism"))
```
or all lower-case...

File: Applications/Utils/SimpleMeshCreation/generateStructuredMesh.cpp [hunk 2]
```diff
@@ -221,6 +224,11 @@ int main (int argc, char* argv[])
     case MeshLib::MeshElemType::HEXAHEDRON:
         mesh.reset(MeshLib::MeshGenerator::generateRegularHexMesh(*vec_div[0], *vec_div[1], *vec_div[2]));
         break;
+    case MeshLib::MeshElemType::PRISM:
+        mesh.reset(MeshLib::MeshGenerator::generateRegularPrismMesh(
+            n_subdivision[0], n_subdivision[1], n_subdivision[2], vec_dx[0],
+            vec_dx[1], vec_dx[2]));
```
Comment (endJunction) [inline Applications/Utils/SimpleMeshCreation/generateStructuredMesh.cpp:230]: Why is `vec_dx` passed here and `vec_div` in other calls?

Comment (TomFischer) [reply Applications/Utils/SimpleMeshCreation/generateStructuredMesh.cpp:230]: There doesn't exist a corresponding interface in `MeshLib/MeshGenerators/MeshGenerator.h`.

Comment (endJunction) [reply Applications/Utils/SimpleMeshCreation/generateStructuredMesh.cpp:230]: https://github.com/ufz/ogs/blob/master/MeshLib/MeshGenerators/MeshGenerator.h#L389-L396

Comment (TomFischer) [reply Applications/Utils/SimpleMeshCreation/generateStructuredMesh.cpp:230]: Okay, you are right. There is an interface that takes the length and the number of subdivisions.

The other generator functions take `BaseLib::ISubdivision` as a parameter. An overload for the generateRegularPrismMesh that takes `BaseLib::ISubdivision` doesn't exist. So I have chosen the above function without a special reason. I'll change it to the suggested function to be a bit more consistent to the other generator functions.

**Comment by bilke** [comment]
@TomFischer  ThirdParty/cmake-modules was changed (accidentally I guess...).

**Comment by TomFischer** [comment]
> @TomFischer ThirdParty/cmake-modules was changed (accidentally I guess...).

Yes, this was an accident. I'll fix it.

**Pull request merged**